OpenRA.Mods.Common/Traits/Render/WithMuzzleOverlay.cs

```diff
@@ -80,6 +80,10 @@ void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barre
 			if (a.Info.MuzzleSplitFacings > 0)
 				sequence += Util.QuantizeFacing(getFacing(), a.Info.MuzzleSplitFacings).ToString();
 
+			// Weapon might had been fired from some other attack base, such as AttackGarrisoned
+			if (!anims.ContainsKey(barrel))
```

Comment:
Can we move this check above the `var sequence = ...;` line? (To save some unnecessary computing in case this condition is met.)